backend/manager/modules/vdsbroker/src/main/java/org/ovirt/engine/core/vdsbroker/VdsManager.java

```diff
@@ -1247,10 +1260,37 @@ public void vmsMonitoringInitFinished() {
         if (!isInitialized()) {
             log.info("VMs initialization finished for Host: '{}:{}'", cachedVds.getName(), cachedVds.getId());
             resourceManager.handleVmsFinishedInitOnVds(cachedVds.getId());
+            updateCpuPinning(vmDao.getMonitoredVmsRunningByVds(cachedVds.getId()));
             setInitialized(true);
         }
     }
 
+    private void updateCpuPinning(List<VM> vms) {
+        for (VM vm : vms) {
+            String pinning;
+            switch (vm.getCpuPinningPolicy()) {
+                case MANUAL:
+                    pinning = vm.getCpuPinning();
+                    break;
+                case RESIZE_AND_PIN_NUMA:
+                case DEDICATED:
+                    pinning = vm.getCurrentCpuPinning();
+                    break;
+                default:
+                    return;
+            }
+            if (pinning != null) {
+                synchronized (this) {
+                    cpuTopology.stream().filter(cpu -> CpuPinningHelper.getAllPinnedPCpus(pinning)
+                            .contains(cpu.getCpu())).forEach(cpu ->  {
```

Comment:
please try to have `cpu -> CpuPinningHelper.getAllPinnedPCpus(pinning).contains(cpu.getCpu())` on the same line - it's hard to read it this way